src/index.ts

```diff
@@ -1462,10 +1462,10 @@ function FlatpickrInstance(
         self.element.contains(eventTarget as HTMLElement) ||
         // web components
         // e.path is not present in all browsers. circumventing typechecks
-        ((e as any).path &&
-          (e as any).path.indexOf &&
-          (~(e as any).path.indexOf(self.input) ||
-            ~(e as any).path.indexOf(self.altInput)));
+        ((e as any).composedPath() &&
+          (e as any).composedPath().indexOf &&
+          (~(e as any).composedPath().indexOf(self.input) ||
+            ~(e as any).composedPath().indexOf(self.altInput)));
```

Comment:
I think this can be simplified, since if `composedPath` exists, it's guaranteed to return an array, which is guaranteed to have an `indexOf` method. Also need a type guard for self.altInput to please Typscript.
```suggestion
       (e.composedPath &&
         (~e.composedPath().indexOf(self.input) ||
           (self.altInput && ~e.composedPath().indexOf(self.altInput))));
```

Let's also remove the second line of the above comment, since it no longer applies.